projects/js-packages/publicize-components/src/components/social-post-control/index.js

```diff
@@ -18,12 +13,24 @@ const SHARE_AS_SOCIAL_POST_TEXT = __(
 /**
  * The component that handles the social post checkbox.
  *
- * @param {boolean} disabled -  Indicates whether the checkbox is disabled or not.
+ * @param {object} props - The properties passed to the component.
+ * @param {boolean} props.disabled -  Indicates whether the checkbox is disabled or not.
+ * @param {boolean} props.isCustomMediaAvailable - Indicates whether custom media is available or not.
  * @returns {object} The SocialPostControl Component.
  */
-export default function SocialPostControl( { disabled = false } ) {
+export default function SocialPostControl( { disabled = false, isCustomMediaAvailable = true } ) {
 	const { shouldUploadAttachedMedia, updateShouldUploadAttachedMedia } = useAttachedMedia();
```

Comment:
I wonder whether we should make the default for Notes to share as a social post, considering the changes we're making to the link format. Having different defaults for different post formats could be really confusing though, and someone might want to change the default. Something to think about!

I think name/label 'Share as a social post' is something we should change. It's interesting that we call it 'upload attached media' here.

Reply:
I agree. It's little confusing.

Reply:
So, which part is confusing? Should I enable it by default? I'm not sure if changing the name/label is a good idea, as it's the same functionality, just for different post types, we want the users to recognize that 🤔

Reply:
>  It's interesting that we call it 'upload attached media' here.

Yeah we could refactor this component a bit in another change to make the naming more clear.